### ISSUE TITLE
What is the right way to set the footnotes font?

### ISSUE THREAD

**Issue by ctrlcctrlv** [sile-typesetter/sile#986]

**Comment by alerque** [comment]
Redefine the command for `\footnote:font`.

```lua
SILE.registerCommand("footnote:font", function (_, content)
  SILE.call("font", { size = "9pt" }, content)
end)
```

**Comment by ctrlcctrlv** [comment]
Hmm, it feels like there should be an option :-)

**Comment by alerque** [comment]
That command (and many similar ones) exist explicitly so there is a place to hook in and set your own styling.

In CaSILE's book class `cabook` I overhauled the namespacing a bit, but the fundamental idea is the same. For example the default footnote font is specified as an inheritance from the "alternate" serif font (which by default just inherits from the default `cabook:font:serif`).

```lua
SILE.registerCommand("cabook:font:footnote", function (options, content)
  options.size = options.size or "0.74em"
  SILE.call("cabook:font:alt", options, content)
end)
```

Because of the way the options are set to be passed up the chain through these functions, you don't have to completely redefine the function to change something (although you *can* still do so). Instead you can just change the default contents of the options table:

```lua
SILE.setCommandDefaults("cabook:font:footnote", { size = "6.5pt", family = "Crimson Pro" })
```

I've been meaning to propose SILE's default classes be overhauled with namespaces and option chaining like this. How does it strike you?

**Comment by ctrlcctrlv** [comment]
Quite well, my use case I think is pretty typical, using EB Garamond for the book; some of my other documents use URW's Palatino. I don't really know of why you'd want footnotes in a different font, so it seems a strange default to be honest.

**Comment by alerque** [comment]
By default they aren't a different font, they are just a different size. I think the problem you are experiencing is that you are changing the activly used font of the current typesetter, but the footnote frame starts a new typesetter and hence reverts back to the default font. Perhaps you should change the default font for the scope of your documents instead, that way headers, footers, folios, footnotes, etc. (anything that starts a new frame or typesetter instance) keeps your font.

From TeXlike input:

```sile
\set[parameter=font.family,value=EB Garamond,makedefault=true]
```

For in Lua:

```lua
-- This is what it *should* be but I just reaziled we never implemented this!
SILE.settings.set("font.family", "EB Garamond", true)

-- This works for now
SILE.settings.defaults["font.family"] = "EB Garamond"
```

**Comment by ctrlcctrlv** [comment]
Indeed, you are correct, I was doing it wrong, using `\font` in the typesetter, and not that `\set`. Works great now. Thanks!

**Comment by alerque** [comment]
I'm glad that fixes this for you, but I'm going to re-open this because I think it is insufficiently documented. I remember being surprised by this early on and I've run into it myself multiple times since. It's bound to come up for other people that don't realize behind the scenes what is actually going on when you change the current font vs. what happens when a new typesetter instance spawns. I would have liked to answer this with "read §x.y" in the manual, but it isn't clearly covered.

**Comment by Omikhleia** [comment]
Stumbling upon that issue after experimenting with #1027 ...

(After some more thought) Besides **footnote*, the more general remark here actually applies to all packages that reset the settings, e.g. also **folios**  and probably - I have not idea what is yet, but also **pecha**).  It's more "What is the right way to set the main document font")

From the existing Manual, I sincerely thought the way to set the document font globally would be to have, just after loading a few packages, the following command (here with a dumb font choice for illustration):

```
\font[family=Ubuntu Mono]
```
It seemed to be the most logical way... But then I realized the footnote font was reverted to the default Gentium:

![image](https://user-images.githubusercontent.com/18075640/132946159-833a4792-f2f8-4734-a000-583c62e82898.png)

Digging through the issues, finding this one... Ok, Let's switch to:
```
\set[parameter=font.family,value=Ubuntu Mono,makedefault=true]
```

Yippee-ya-yay, it of works ! 

But in my previous documents, I hadn't noticed that fact, due to the fonts being so close...

Of course the Manual mentions the settings and the `makedefault` option, but it comes pretty late (Chapter 7), within complex settings (orphanpenalty, spacing, etc.) which are pretty low-level compared to the initial point, for a new user, which is just to tell SILE which font to use! One is not going to read this section just to get a basic grasp the software.

Anyway ok, let's say it's a documentation issue... But then it should likely be explained very soon in the Manual. I'm pretty sure any user would want to set his preferred font and expect it to be applied?

But... It starts felling apart again when I  said to myself "I don't know yet if I'll typeset in 10pt or 12pt eventually, let's make the footnote font relative, so it adapts by some ratio" - by the way, the hard-coded 9pt default is questionable anyway...

Hereby:
```
SILE.registerCommand("footnote:font", function (_, content)
  print("note: "..SILE.settings.get("font.size"))
  SILE.call("font", { size = SILE.settings.get("font.size") * 0.8 }, function ()
    SILE.process(content)
  end)
end)
}
```

Ahh, but of course, the toplevel default is 10pt, this is not going to work as-is... And I now know that a `\font[family=Ubuntu Mono, size=12pt]` wouldn't do the trick, but that I should add something to the above solution:

```
\set[parameter=font.family,value=Ubuntu Mono,makedefault=true]%
\set[parameter=font.size,value=12,makedefault=true]% Warning, it takes a number (12), not a length specification (12pt)
```

Er.. .. And likewise, I guess - although it gets borderline - if I also want to select some OpenType features.. (but why not, actually, just to configure some ligatures and stylistic set, &c). Another `\set` will do... Doh!

_To make a long story short_,
- At least, "Setting the global document font" should be documented _early_ in the manual.
- But it still seems somehow wrong, nevertheless, to impose all those low-level `\set` commands... I wonder if there wouldn't be an easier way:
   -  Such as a `\fontdefault` command, with same syntax as `font` ? Looks like a dirty workaround though...
   - Or the settings should not be reset to initial values there (footnotes, folios, &c), but maybe they could instead be set to what the very first user-document state had. It must be somewhere in the settings stack/queue, right?
